pyincore/semanticservice.py

```diff
@@ -0,0 +1,178 @@
+            **kwargs: Arbitrary keyword arguments.
+
+        Returns:
+            list: List of semantic types.
+
+        """
+
+        url = self.base_url
+        payload = {
+            "hyperlink": hyperlink,
+            "order": order,
+            "skip": skip,
+            "limit": limit,
+            "detail": detail,
+        }
+        response = self.client.get(
+            url, params=payload, timeout=timeout, **kwargs
+        )
+        data = self.return_http_response(response).json()
+
+        if save_json:
+            # if json_path is None: then save to the default location (incore cache folder)
+            if json_path is None:
+                json_path = os.path.join(
+                    pyglobals.PYINCORE_USER_CACHE,
+                    f"semantic_types_ord_{order}_sk_{skip}_lim_{limit}.json",
+                )
+            with open(json_path, "w") as f:
+                json.dump(data, f, indent=4)
+
+        return data
+
+    def get_semantic_type_by_name(
+        self,
+        type_name: str,
+        save_json: bool = False,
+        json_path: str = None,
+        timeout: Tuple[int, int] = (30, 600),
+        **kwargs,
+    ) -> Union[list, dict]:
+        """Get semantic type by name.
+
+        Args:
+            type_name (str): Name of the semantic type.
+            save_json (bool): If True, save the json response to a file.
+            json_path (str): Path to save the json response.
+            timeout (tuple): Timeout for the request in seconds.
+                            First value is the connection timeout, second value is the read timeout.
+            **kwargs: Arbitrary keyword arguments.
+
+        Returns:
+            list or dict: List or dictionary of semantic type.
+
+        """
+
+        url = f"{self.base_url}/{type_name}"
+        response = self.client.get(url, timeout=timeout, **kwargs)
+        data = self.return_http_response(response).json()
+
+        if save_json:
+            # if json_path is None: then save to the default location (incore cache folder)
+            if json_path is None:
+                json_path = os.path.join(
+                    pyglobals.PYINCORE_USER_CACHE, f"{type_name}.json"
+                )
+            with open(json_path, "w") as f:
+                json.dump(data, f, indent=4)
+
+        return data
+
+    def search_semantic_type(
+        self, query: str, timeout: Tuple[int, int] = (30, 600), **kwargs
+    ) -> Union[list, dict]:
+        """Search for a semantic type.
+
+        Args:
+            query (str): Query to search for.
+            timeout (tuple): Timeout for the request in seconds.
+                            First value is the connection timeout, second value is the read timeout.
+            **kwargs: Arbitrary keyword arguments.
+
+        Returns:
+            list or dict: List or dictionary of semantic type.
+
+        """
+
+        url = self.base_url + "/search"
```

Comment:
Could we use url = urljoin(self.base_url, "search") instead?

Reply:
<img width="568" alt="Screenshot 2023-07-26 at 4 14 04 PM" src="https://github.com/IN-CORE/pyincore/assets/32737711/1a191daf-4e0a-4ca7-8b08-a6da2b42135b">
This is the error I get. As you can see in the very next line I print the urljoin output.... and it only contains the other second parameter... Idk it just omits the first part